lib/models/NotificationConfiguration.js

```diff
@@ -0,0 +1,245 @@
+                    filterRules: filterObj.filterRules,
+                });
+            }
+        });
+        return parseError;
+    }
+
+    /**
+     * Check that events array is valid
+     * @param {array} events - event array
+     * @return {object} - contains error if parsing failed or events array
+     */
+    _parseEvents(events) {
+        const eventsObj = {
+            events: [],
+        };
+        if (!events || !events[0]) {
+            eventsObj.error = errors.MalformedXML.customizeDescription(
+                'each queue configuration must contain an event');
+            return eventsObj;
+        }
+        events.forEach(e => {
+            if (!supportedNotificationEvents.includes(e)) {
+                eventsObj.error = errors.MalformedXML.customizeDescription(
+                    'event array contains invalid or unsupported event');
+            }
+            eventsObj.events.push(e);
+        });
+        return eventsObj;
+    }
+
+    /**
+     * Check that filter array is valid
+     * @param {array} filter - filter array
+     * @return {object} - contains error if parsing failed or filter array
+     */
+    _parseFilter(filter) {
+        const filterObj = {
+            filterRules: [],
+        };
+        if (filter && filter[0]) {
+            if (!filter[0].S3Key || !filter[0].S3Key[0]) {
+                filterObj.error = errors.MalformedXML.customizeDescription(
+                    'if included, queue configuration filter must contain S3Key');
+                return filterObj;
+            }
+            const filterRules = filter[0].S3Key[0];
+            if (!filterRules.FilterRule || !filterRules.FilterRule[0]) {
+                filterObj.error = errors.MalformedXML.customizeDescription(
+                    'if included, queue configuration filter must contain a rule');
+                return filterObj;
+            }
+            const ruleArray = filterRules.FilterRule;
+            ruleArray.forEach(r => {
+                if (!r.Name || !r.Name[0] || !r.Value || !r.Value[0]) {
+                    filterObj.error = errors.MalformedXML.customizeDescription(
+                        'each included filter must contain a name and value');
+                } else if (!['Prefix', 'Suffix'].includes(r.Name[0])) {
+                    filterObj.error = errors.MalformedXML.customizeDescription(
+                        'filter Name must be one of Prefix or Suffix');
+                } else {
+                    filterObj.filterRules.push({
+                        name: r.Name[0],
+                        value: r.Value[0],
+                    });
+                }
+            });
+        }
+        return filterObj;
+    }
+
+    /**
+     * Check that id string is valid
+     * @param {string} id - id string (optional)
+     * @return {object} - contains error if parsing failed or id
+     */
+    _parseId(id) {
+        const idObj = {};
+        idObj.propName = 'ruleID';
```

Comment:
Can initialize in one.
```suggestion
        const idObj = { propName: 'ruleID' };
```